catroid/src/main/java/org/catrobat/catroid/ui/recyclerview/adapter/CategoryListRVAdapter.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Catroid: An on-device visual programming system for Android devices
+ * Copyright (C) 2010-2018 The Catrobat Team
+ * (<http://developer.catrobat.org/credits>)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * An additional term exception under section 7 of the GNU Affero
+ * General Public License, version 3, is available at
+ * http://developer.catrobat.org/license_additional_term
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.catrobat.catroid.ui.recyclerview.adapter;
+
+import android.support.annotation.LayoutRes;
+import android.support.annotation.Nullable;
+import android.support.v7.widget.RecyclerView;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+
+import org.catrobat.catroid.R;
+import org.catrobat.catroid.ui.recyclerview.viewholder.CategoryListItemVH;
+
+import java.util.List;
+
+public class CategoryListRVAdapter extends RecyclerView.Adapter<CategoryListItemVH> {
+
+	public enum CategoryListItemType {
+		DEFAULT, COLLISION, NXT, EV3
+	}
+
+	public static class CategoryListItem {
+		@Nullable
+		public String header;
+		public int nameResId;
+		public String text;
+		public CategoryListItemType type;
+
+		public CategoryListItem(int nameResId, String text, CategoryListItemType type) {
+			this.nameResId = nameResId;
+			this.text = text;
+			this.type = type;
+		}
+	}
+
+	private List<CategoryListItem> items;
+	private OnItemClickListener onItemClickListener;
+
+	public CategoryListRVAdapter(List<CategoryListItem> items) {
+		this.items = items;
+	}
+
+	@Override
+	public CategoryListItemVH onCreateViewHolder(ViewGroup parent, int viewType) {
+		View view = LayoutInflater.from(parent.getContext()).inflate(viewType, parent, false);
+		return new CategoryListItemVH(view);
+	}
+
+	@Override
+	public void onBindViewHolder(CategoryListItemVH holder, final int position) {
+		if (items.get(position).header != null) {
```

Comment:
instead of checking if the header is null you can use items.get(position).getItemViewType() and then set the values accordingly.